src/app/src/components/DrawTools/DrawTools.js

```diff
@@ -9,7 +12,27 @@ import useEditingPolygon from './useEditingPolygon';
 import useGeocoderResult from './useGeocoderResult';
 import useTrackMapZoom from './useTrackMapZoom';
 
-export default function DrawTools() {
+import { addPolygon } from '../../store/mapSlice';
+
+export default function DrawTools({ details }) {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (details) {
+            dispatch(
+                addPolygon({
```

Comment:
Unrelated to your changes, but at first I was worried this would add multiple polygons (since details is an object and won't memoize accross renders), but it turns out `addPolygon` is overriding the polygon in state anyway. Maybe this would be better if named `setPolygon`?